templates/interfaces.j2

```diff
@@ -1,49 +1,130 @@
+
+{% elif bridge.type|default(network_management_default_type) == "static" or
+   bridge.type|default(network_management_default_type) == "manual" -%}
+
+{% for ip in bridge.ips %}{# start: additional ips #}
+	{% if not loop.first %}
 
+iface {{ bridge.name }} inet {{ bridge.type|default(network_management_default_type) }}
+	{% endif -%}
+	address {{ ip | ipaddr('address') }}
+	network {{ ip | ipaddr('network') }}
+	netmask {{ ip | ipaddr('netmask') }}
+	broadcast {{ bridge.broadcast|default(ip | ipaddr('broadcast')) }}
+{% if bridge.gateway is defined or network_management_default_gateway is defined %}
+	gateway {{ bridge.gateway|default(network_management_default_gateway )|ipaddr('address') }}
 {% endif %}
-{% else %}
-iface {{ interface.name }} inet manual
+{% endfor %}{# end: additional ips #}
 {% endif %}
-	pre-up    ovs-vsctl --may-exist add-br {{ interface.name }} {% if interface.parent is defined %} {{ interface.parent }} {{ interface.vlan }} {% endif %}
+	dns-nameservers{% for nameserver in bridge.nameservers|default(network_management_nameservers) %} {{ nameserver|ipaddr('address') }}{% endfor %}{{ '\n' }}{# for loop, to allow ipaddr filter usage #}
+{% for port in bridge.ports | default({}) %}
+	post-down ovs-vsctl --if-exists del-port {{ bridge.name }} {{ port.port }}
+{% endfor %}
+	post-down ovs-vsctl --if-exists del-br {{ bridge.name }}
 
-{% for port in interface.ports | default({}) %}
-	post-up   ovs-vsctl --may-exist add-port {{ interface.name }} {{ port.port }} {% if port.vlan is defined %} tag={{ port.vlan }} {% endif %}
 
 {% endfor %}
-{% for port in interface.ports | default({}) %}
-	pre-down  ovs-vsctl --if-exists del-port {{ interface.name }} {{ port.port }}
-{% endfor %}
-	post-down ovs-vsctl --if-exists del-br {{ interface.name }}
+{% if patch_field is defined %}
 
+################################################################################
+# OpenVswitch Patchfield, connect requested ports together                     #
+################################################################################
+{% for patch_field_dev_in, patch_field_dev_out in patch_field.iteritems() %}
+post-up ovs-vsctl {{''
+	-}} --may-exist add-port {{ patch_field_dev_in }} {{ patch_field_dev_in }}_patch_{{ loop.index }} -- {{''
+	-}} set interface {{ patch_field_dev_in }}_patch_{{ loop.index }} type=patch options:peer={{ patch_field_dev_out }}_patch_{{ loop.index }} -- {{''
+	-}} --may-exist add-port {{ patch_field_dev_out }} {{ patch_field_dev_out }}_patch_{{ loop.index }} -- {{''
+	-}} set interface {{ patch_field_dev_out }}_patch_{{ loop.index }} type=patch options:peer={{ patch_field_dev_in }}_patch_{{ loop.index }}; true
 {% endfor %}
+{% endif %}{# end: if patch_field #}
+{% endif %}{# end: if bridges #}
+{% if network_management_default_gateway is defined %}
+
+
+post-up ip r replace default via {{ network_management_default_gateway }}
```

Comment:
Are you sure this is the correct functionality?
If I specify a gateway on an interface it will be replaced by the default gatway. Shouldn't the default gatway only be used if an interface does not specify a gatway.
I may be wrong here though.

Reply:
Yup, I'm only the default route is replaced, no interface specific routes.